src/test/java/org/openmrs/contrib/qaframework/automation/EncountersSteps.java

```diff
@@ -0,0 +1,101 @@
+package org.openmrs.contrib.qaframework.automation;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.reference.page.AdmitToInpatientPage;
+import org.openmrs.reference.page.PatientVisitsDashboardPage;
+import org.openmrs.uitestframework.test.TestData;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public class EncountersSteps extends Steps {
+
+	private TestData.PatientInfo testPatient;
+
+	@Before(RunTest.HOOK.SELENIUM_ENCOUNTER)
+	public void visitHomePage() {
+		testPatient = createTestPatient();
+		initiateWithLogin();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_ENCOUNTER)
+	public void destroy() {
+		deletePatient(testPatient);
+		quit();
+	}
+
+	@Given("a user clicks on Find Patient app from Home page")
+	public void goToFindPatientRecordPage() {
+		findPatientPage = homePage.goToFindPatientRecord();
+	}
+
+	@Then("the system loads find patient record page")
+	public void systemloadsFindPatientRecordPage() {
+		assertTrue(textExists("Find Patient Record"));
+	}
+
+	@When("a user searches an existing patient and starts the visit")
+	public void searchExistingPatient() {
+		findPatientPage.enterPatient(testPatient.identifier);
+		findPatientPage.waitForPageToLoad();
+		visitsDashboardPage = findPatientPage.clickOnFirstPatient()
+				.startVisit();
+	}
+
+	@Then("the system loads patient visit dashboard page")
+	public void systemloadsVisitDashboardPage() {
+		assertTrue(visitsDashboardPage.containsText("Visits"));
+	}
+
+	// Admission encounter
+	@When("a user clicks on Admit to Inpatient button and selects location")
+	public void goToAdmitToInpatientPage() {
+		visitsDashboardPage.goToAdmitToInpatient().confirm("Inpatient Ward");
```

Comment:
It would have been nice to have them separated but from the design of the page, the method confirm can only be accessed using the method goToAdmitToInpatient() other than that the method can not be accessible. I found out the issue are originating from the design of the page as observed [here](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/f2e3df96f7ff815ee8249c4dce61ae460339f770/ui-tests/src/test/java/org/openmrs/reference/page/TransferPatientPage.java#L23).

Refactoring the page will by default fail the selenium tests that consume the function and personally would not prefer taking time to dive into refactoring the page but rather focus on writing tests for functionalities that were not covered by the selenium test.

Yes if we are to look into refactoring the[ page](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/f2e3df96f7ff815ee8249c4dce61ae460339f770/ui-tests/src/test/java/org/openmrs/reference/page/TransferPatientPage.java#L6) then we can create a ticket to cater for  that and would be good for new contributors to kick off with such tickets But this should not put this test in a waiting state!

Reply:
> Yes if we are to look into refactoring the[ page](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/f2e3df96f7ff815ee8249c4dce61ae460339f770/ui-tests/src/test/java/org/openmrs/reference/page/TransferPatientPage.java#L6) then we can create a ticket to cater for that and would be good for new contributors to kick off with such tickets But this should not put this test in a waiting state!

This is fine we can leave this for now